test/runner.js

```diff
@@ -1812,6 +1812,78 @@ describe('Runner', () => {
         expect(notebook.failures).to.equal(1);
     });
 
+    it('passes shallow clones of context to tests', async () => {
+
+        const script = Lab.script({ schedule: false });
+
+        const testContext = {
+            testData: { hello: 'there' }
+        };
+
+        const innerContext = {
+            testData: { goodbye: 'you' },
+            additionalData: { another: 'object' }
+        };
+
+        script.experiment('test', () => {
+
+            script.before(({ context }) => {
+
+                context.testData = testContext.testData;
+            });
+
+            script.test('has test context', ({ context }) => {
+
+                expect(context,'has proper context').to.equal(testContext);
+                expect(context,'is a shallow clone').to.not.shallow.equal(testContext);
+                expect(context.testData,'is a reference').to.shallow.equal(testContext.testData);
+                context.anotherProperty = { something: 'random' };
+            });
+
+            script.test('does not see changes to context from previous test', ({ context }) => {
+
+                expect(context,'has proper context').to.equal(testContext);
+                expect(context,'is a shallow clone').to.not.shallow.equal(testContext);
+                expect(context.testData,'is a reference').to.shallow.equal(testContext.testData);
+                expect(context.anotherProperty,'uses the original clone').to.not.exist();
+            });
+
+            script.experiment('inner test', () => {
+
+                script.before(({ context }) => {
+
+                    context.testData = innerContext.testData;
```

Comment:
I would be interested to see if the `context` passed here shallow equals (`===`) the one passed to the outer experiment's `before()` block.

Reply:
i was trying to keep assertions out of the before and after blocks since it seems to cause some weirdness.. do you not think the outer block's test immediately before this block is defined is enough?

Reply:
Hmm, good question.  What I'm picturing would look like this:
```js
            let beforeContext;
            script.before(({ context }) => {

                beforeContext = context;
                context.testData = testContext.testData;
            });

            // ...

                let innerBeforeContext;
                script.before(({ context }) => {

                    innerBeforeContext = context;
                    context.testData = innerContext.testData;
                });

                script.test('context is not shallow-cloned across before()s', ({ context }) => {

                    expect(beforeContext,'is a reference').to.shallow.equal(innerBeforeContext);
                });
```